__tests__/frontend/Recommend-test.js

```diff
@@ -0,0 +1,109 @@
+import axios from 'axios'
+import React from 'react'
+import MockAdapter from 'axios-mock-adapter'
+import { render } from '@testing-library/react-native'
+import { ENDPOINTS } from '../../src/api/playlists'
+import { HTTP_STATUS_CODES } from '../../src/api'
+import {
+  PLAYLISTS_BY_OFFERING_RESPONSE,
+  VIDEOS_BY_PLAYLIST_RESPONSE,
+} from '../mock_responses/mock-playlists-response'
+import { format } from '../../src/utils/string'
+import Reccomend from '../../src/components/Recommend/Recommend'
+
+const mock = new MockAdapter(axios)
+describe('Check videos rendering -- mode false', () => {
+  const offeringId = 'ac5b1727-629c-443b-8c1a-cc1bd541af6a'
+  const playlistId = '51519746-aa6c-485c-9894-549959c457b5'
+
+  afterEach(() => {
+    mock.reset()
+  })
+
+  test('when given many Videos', async () => {
+    const mockNaivgator = { push: jest.fn() }
+    mock
+      .onGet(`${format(ENDPOINTS.PLAYLISTS_BY_OFFERING, offeringId)}`)
+      .reply(HTTP_STATUS_CODES.OK, PLAYLISTS_BY_OFFERING_RESPONSE)
+      .onGet(`${format(ENDPOINTS.VIDEOS_BY_PLAYLIST, playlistId)}`)
+      .reply(HTTP_STATUS_CODES.OK, VIDEOS_BY_PLAYLIST_RESPONSE)
+
+    const { queryByText } = render(
+      <Reccomend courseId={offeringId} navigation={mockNaivgator} mode={false} />
+    )
+    expect(queryByText('Placeholder')).toBe(null)
```

Comment:
These tests will actually be flaky because when you render `<Reccomend />` nothing is rendered until you make the endpoint call in `useEffect`. So the component is first rendered with nothing, then once the endpoint returns data, it is rendered again with the videos. So if your tests run after rendering the component but before making the endpoint call, then it will fail.

To get around this, wrap all of your `queryByText` with `waitFor`. This will wait for the item to be rendered or time out. So for example, line 34 becomes, `expect( await waitFor(() => queryByText('Placeholder'))).toBe(null)`. You could always break it up into mulitple lines if it gets to messy.